### PR TITLE
parallelise the code-gen

### PR DESCRIPTION
**Pull request by danieleades**
inspired by @GabrielDertoni's work in #161, this PR provides a (much more modest) performance boost by parallelising the code-gen using rayon

Screenshot 2023-03-21 080654

### CODE REVIEW

**Comment by danieleades** [comment]
- does rayon provide a speed up on all targets, or only some?
- is it worth feature-gating this?

**Comment by GabrielDertoni** [comment]
I would like to see how this does after #161 gets merged. Maybe the 1.84s of rayon comp time will already make it not worth it.

File: Cargo.toml [hunk 1]
```diff
@@ -51,22 +52,7 @@ heapless = "0.7"
...
-    "python_array_test",
-    "slugs",
-    "standard",
-    "test",
-    "ualberta",
-    "uavionix",
-    "icarous",
-    "common",
-]
+"all-dialects" = ["ardupilotmega", "asluav", "autoquad", "matrixpilot", "minimal", "paparazzi", "python_array_test", "slugs", "standard", "test", "ualberta", "uavionix", "icarous", "common"]
```
Comment (patrickelectric) [inline Cargo.toml]: do not follow up style changes with funcional changes as well, having such list in a huge single line is harder to maintain.

**Review by patrickelectric — CHANGES_REQUESTED** [review]
Just some unnecessary style change to approve this, everything else looks awesome.

**Comment by danieleades** [comment]
> I would like to see how this does after #161 gets merged. Maybe the 1.84s of rayon comp time will already make it not worth it.

I agree it's worth merging your changes first and then reassessing.

I think also it maybe be worth considering initial compilation and subsequent compilation separately. For example maybe its worth compiling rayon *once* to get a benefit on every subsequent recompilation of this library

File: Cargo.toml (hunk 1 above)
Comment (danieleades) [reply Cargo.toml]: no problem. This was introduced as an artefact of using `cargo edit` to add the rayon dependency (which can change formatting), not an intentional change

**Comment by GabrielDertoni** [comment]
Another way to significantly improve developer experience for this crate is using a separate crate for codegen, and letting it run `build/parse.rs` and including the generated code. This way, developers that don't want to change the parser, don't need to wait for the build script to rerun and all the exact same generated code get compiled once again.

**Pull request closed without merging**